demos/demos_by_use_case/bio/cpu/metagenomic_before_gpu.ipynb

```diff
@@ -0,0 +1,721 @@
+            "@SRR9224006.2 7001174F:HVTFNBCXX161011:HVTFNBCXX:1:1108:17623:49640/1\n",
+            "AAAAAAAACAACCCAATGCGATTCTGATCGCAATCTACATAAGTTACTACTGGTTATCTTCCCTGAAG\n",
+            "+\n",
+            "DDBDDIGIIHIIIGHGHFHIGHIIIEHIHIIIIIIIIIIIIEHHHGHIIIIIIIIIIIIIIEHHHHHH\n",
+            "@SRR9224006.3 7001174F:HVTFNBCXX161011:HVTFNBCXX:1:1204:5345:82516/1\n",
+            "AAAAAAAACAAGAGCTTTATTAAACACGTCTTGATCTTTTTTACACCTGCCGGAAATTCCATCGT\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from Bio import SeqIO\n",
+        "import glob,os\n",
+        "import pandas as pd\n",
+        "B=pd.DataFrame()\n",
+        "for i in glob.glob('/content/*.fastq'):\n",
+        "    # j=os.path.basename(i)\n",
+        "    fasta_sequences = SeqIO.parse(open(i),'fastq')\n",
+        "    identifiers = []\n",
+        "    sequences = []\n",
+        "    for fasta in fasta_sequences:\n",
+        "        name, sequence = fasta.id, str(fasta.seq)\n",
+        "        identifiers.append(name)\n",
+        "        sequences.append(sequence)\n",
+        "\n",
+        "    A=pd.DataFrame([identifiers,sequences]).T\n",
+        "    A.columns=['ID','seq']\n",
+        "    A.dropna(inplace=True)\n",
+        "    B=B.append(A)\n",
+        "    # A['ID']#=A.ID.str.split('-')[0:1]\n",
+        "# B['ID']=B['ID'].str.split('-').str[0]+'_'+B['ID'].str.split('-').str[1]#.cat()\n",
+        "B['ID']=B.ID.str.split('_length').str[0]\n",
+        "B.index=B.ID"
+      ],
+      "metadata": {
+        "id": "UWDGFCWxSpv_"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# install [HUMAnN 3](https://huttenhower.sph.harvard.edu/humann), a method for efficiently and accurately profiling the abundance of microbial metabolic pathways and other molecular functions from metagenomic or metatranscriptomic sequencing data."
+      ],
+      "metadata": {
+        "id": "fDvymRFjHPNB"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# !pip install humann --no-binary :all:\n",
+        "!pip install metaphlan"
+      ],
+      "metadata": {
+        "id": "yIeeDXPBHN6D"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "### !humann_databases --download utility_mapping full /path/to/databases --update-config yes\n",
+        "\n",
+        "# !humann_test\n",
+        "\n",
+        "# !wget https://github.com/biobakery/humann/raw/master/examples/demo.fastq.gz\n",
+        "# !humann -i demo.fastq.gz -o sample_results"
+      ],
+      "metadata": {
+        "id": "TnYmqPJ-bHPa"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### takes very long for running all samples\n",
```

Comment:
put the file somewhere (if big, not here), and use a conditional for whether to recompute, and by default, just fetch